src/syslog-counter/src/lib.rs

```diff
@@ -0,0 +1,134 @@
+use std::io::{self, BufRead, Write};
+use std::sync::{Arc, Mutex};
+use std::thread;
+use std::time::Duration;
+use log::{error, info};
+use std::process::exit;
+use swss_common::*;
+
+pub fn convert_to_CxxString(opt: Option<u64>) -> Option<CxxString> {
+    opt.map(|num| CxxString::new(num.to_string()))
+}
+
+pub fn convert_to_u64(opt: Option<&CxxString>) -> Option<u64> {
+    opt.and_then(|s| match s.to_str() {
+        Ok(text) => text.parse::<u64>().ok(),
+        Err(_) => None,
+    })
+}
+
+pub trait DbConnectorTrait: Send  {
+    fn new_named(
+        db_name: impl Into<String>,
+        is_tcp_conn: bool,
+        timeout_ms: u32,
+    ) -> Result<Box<dyn DbConnectorTrait + Send + 'static>>
+    where
+        Self: Sized;
+
+    fn hget(&self, key: &str, field: &str) -> Result<Option<CxxString>>;
+    fn hset(&self, key: &str, field: &str, value: &CxxStr) -> Result<()>;
+}
+
+impl DbConnectorTrait for DbConnector {
+    fn new_named(
+        db_name: impl Into<String>,
+        is_tcp_conn: bool,
+        timeout_ms: u32,
+    ) -> Result<Box<dyn DbConnectorTrait + Send + 'static>> {
+        DbConnector::new_named(db_name, is_tcp_conn, timeout_ms)
+            .map(|conn| Box::new(conn) as Box<dyn DbConnectorTrait + Send + 'static>)
+    }
+
+    fn hget(&self, key: &str, field: &str) -> Result<Option<CxxString>> {
+        self.hget(key, field)
+    }
+
+    fn hset(&self, key: &str, field: &str, value: &CxxStr) -> Result<()> {
+        self.hset(key, field, value)
+    }
+}
+
+pub struct CountersDb {
+    pub connector: Box<dyn DbConnectorTrait + Send + 'static>,
+}
+
+impl CountersDb {
+    pub fn new(connector: Box<dyn DbConnectorTrait + Send + 'static>) -> Self {
+        Self { connector }
+    }
+
+    pub fn get_count(&self) -> Result<u64, String> {
+        let raw = self.connector.hget("SYSLOG_COUNTER", "COUNT")
+            .map_err(|e| format!("Failed to hget COUNT: {:?}", e))?;
+        convert_to_u64(raw.as_ref())
+            .ok_or_else(|| "Failed to parse COUNT as u64".to_string())
```

Comment:
```suggestion
        self.connector.hget("SYSLOG_COUNTER", "COUNT")
            .map_err(|e| format!("Failed to hget COUNT: {:?}", e))?;
            .and_then(|u| convert_to_u64(u))
```

This assumes `convert_to_u64` is modified.